src/OrchardCore/OrchardCore.Infrastructure/Documents/VolatileDocumentManager.cs

```diff
@@ -55,9 +55,16 @@ public Task UpdateAtomicAsync(Func<Task<TDocument>> updateAsync)
                 }
 
                 await using var acquiredLock = locker;
-                var updated = await _updateDelegateAsync();
-                updated.Identifier ??= IdGenerator.GenerateId();
-                await SetInternalAsync(updated);
+
+                TDocument document = null;
+                foreach (var d in _updateDelegateAsync.GetInvocationList())
+                {
+                    document = await ((UpdateDelegate)d)();
+                }
+
+                document.Identifier ??= IdGenerator.GenerateId();
```

Comment:
I didn't know the null-coalescing assignment operator `??=`:
https://docs.microsoft.com/fr-fr/dotnet/csharp/language-reference/operators/null-coalescing-operator

Reply:
Yes, we already use it in some places